File: exercise/C#/day19/Travel/SantaTravelCalculator.cs
namespace Travel
{
public static class SantaTravelCalculator
{
    public static int CalculateTotalDistanceRecursively(int numberOfReindeers)
    {
        if (numberOfReindeers == 1) return 1;
        checked
        {
            return 2 * CalculateTotalDistanceRecursively(numberOfReindeers - 1) + 1;
        }
    }
}
}

File: exercise/java/day19/src/main/java/travel/SantaTravelCalculator.java
package travel;

public class SantaTravelCalculator {
    private SantaTravelCalculator() {
    }

    public static int calculateTotalDistanceRecursively(int numberOfReindeers) throws ArithmeticException {
        if (numberOfReindeers == 1) return 1;
        var result = 2 * calculateTotalDistanceRecursively(numberOfReindeers - 1) + 1;
        if (result == -1) throw new ArithmeticException("Integer overflow");

        return result;
    }
}